### PR TITLE
built in help for admin

### PR DESCRIPTION
**Pull request by scottcwilson**
Works on Admin > Localization > Currencies and Admin > Catalog > Categories/Products.  

Help function could be overridden to open a new page with locally created content as well as default content.

### CODE REVIEW

**Comment by scottcwilson** [comment]
image

**Comment by wiztechinc** [comment]
Good start. So does use the new docs for the tips.

**Comment by drbyte** [comment]
Suggestions (related to code, not UI):
- instead of functions/general, I suggest functions/functions_help.php
- instead of database table, a PHP array would be sufficient (I prefer to avoid db tables that are strictly static data, unless they're useful in query joins with other relational data)

**Comment by scottcwilson** [comment]
Roger that.

File: admin/includes/functions/functions_help.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+/**
+ * @package admin
+ * @copyright Copyright 2020-2020 Zen Cart Development Team
+ * @copyright Portions Copyright 2003 osCommerce
+ * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
+ * @version $Id: DrByte 2019 May 26 Modified in v1.5.6b $
+ */
+
+ function page_has_help(&$help_page) {
+   global $db;
```
Comment (drbyte) [inline admin/includes/functions/functions_help.php:11]: $db not needed.

File: admin/includes/functions/functions_help.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * @package admin
+ * @copyright Copyright 2020-2020 Zen Cart Development Team
+ * @copyright Portions Copyright 2003 osCommerce
+ * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
+ * @version $Id: DrByte 2019 May 26 Modified in v1.5.6b $
+ */
+
+ function page_has_help(&$help_page) {
```
Comment (drbyte) [inline admin/includes/functions/functions_help.php:10]: Let's avoid the `&` pass-by-reference here.

File: admin/includes/functions/functions_help.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+     FILENAME_MEDIA_MANAGER => "https://docs.zen-cart.com/user/admin_pages/extras/media_manager/",  
+     FILENAME_MEDIA_TYPES => "https://docs.zen-cart.com/user/admin_pages/extras/media_types/",  
+     FILENAME_STATS_SALES_REPORT_GRAPHS => "https://docs.zen-cart.com/user/admin_pages/reports/graphical_sales_report/",  
+   ); 
+
+   if (isset($pagelist[$page])) {
+     $help_page = $pagelist[$page]; 
+     return true; 
+   }
+   return false;
```
Comment (drbyte) [inline admin/includes/functions/functions_help.php:91]: Returning the URL or `false` will be enough for detecting valid response.

File: admin/includes/header_navigation.php [hunk 4]
```diff
@@ -43,3 +43,12 @@
     </ul>
   </div><!-- /.navbar-collapse -->
 </nav>
+<?php if (page_has_help($help_page)) { ?>
```
Comment (drbyte) [inline admin/includes/header_navigation.php:46]:
```diff
-<?php if (page_has_help($help_page)) { ?>
+<?php if ($url = page_has_help($help_page)) { ?>
```
and then `echo $url` later

File: admin/includes/functions/functions_help.php [hunk 5]
```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * @package admin
+ * @copyright Copyright 2020-2020 Zen Cart Development Team
+ * @copyright Portions Copyright 2003 osCommerce
```
Comment (drbyte) [inline admin/includes/functions/functions_help.php:5]: None of the logic here came from or had inspiration from osc

**Comment by drbyte** [comment]
Thanks for pulling all this together @scottcwilson

**Pull request merged**